ui/packages/catalog-ui-search/src/main/webapp/react-component/location/point-radius.js

```diff
@@ -12,181 +12,300 @@
+    if (type === 'blur') {
+      setDmsError({
+        error: value !== undefined && value.length === 0,
+        message,
+        defaultValue,
+      })
+    } else if (defaultValue) {
+      setDmsError({
+        error,
+        message,
+        defaultValue,
+      })
+    }
+    defaultValue ? setState(key, defaultValue) : setState(key, value)
   }
+
   return (
     <div>
-      <TextField label="USNG / MGRS" value={usng} onChange={cursor('usng')} />
-      <Units value={radiusUnits} onChange={cursor('radiusUnits')}>
-        <TextField label="Radius" value={radius} onChange={cursor('radius')} />
+      <DmsLatitude
+        label="Latitude"
+        value={dmsLat}
+        onChange={(value, type) => validate('dmsLat', type, value)}
+      >
+        <DirectionInput
+          options={latitudeDirections}
+          value={dmsLatDirection}
+          onChange={value => setState('dmsLatDirection', value)}
+        />
+      </DmsLatitude>
+      <DmsLongitude
+        label="Longitude"
+        value={dmsLon}
+        onChange={(value, type) => validate('dmsLon', type, value)}
+      >
+        <DirectionInput
+          options={longitudeDirections}
+          value={dmsLonDirection}
+          onChange={value => setState('dmsLonDirection', value)}
+        />
+      </DmsLongitude>
+      {getErrorComponent(dmsError)}
+      <Units
+        value={radiusUnits}
+        onChange={value => setState('radiusUnits', value)}
+      >
+        <TextField
+          label="Radius"
+          type="number"
+          value={String(radius)}
+          onChange={value => {
+            setRadiusError(validateGeo('radius', value))
+            setState('radius', value)
+          }}
+        />
       </Units>
-      {error ? (
-        <ErrorBlock>
-          <WarningIcon className="fa fa-warning" />
-          <span>Invalid USNG / MGRS coords</span>
-        </ErrorBlock>
-      ) : null}
+      {getErrorComponent(radiusError)}
+    </div>
+  )
+}
+
+const PointRadiusUsngMgrs = props => {
+  const { usng, radius, radiusUnits, setState } = props
+  const [usngError, setUsngError] = useState({ error: false, message: '' })
+  const [radiusError, setRadiusError] = useState({ error: false, message: '' })
+  return (
+    <div>
+      <TextField
+        label="USNG / MGRS"
+        value={usng}
+        onChange={value => setState('usng', value)}
+        onBlur={() => setUsngError(validateGeo('usng', usng))}
+      />
+      {getErrorComponent(usngError)}
+      <Units
+        value={radiusUnits}
+        onChange={value => setState('radiusUnits', value)}
+      >
+        <TextField
+          label="Radius"
+          value={String(radius)}
+          onChange={value => {
+            setRadiusError(validateGeo('radius', value))
+            setState('radius', value)
+          }}
+        />
+      </Units>
+      {getErrorComponent(radiusError)}
     </div>
   )
 }
 
 const PointRadiusUtmUps = props => {
-  const {
+  let {
```

Comment:
❓ why the switch from const to let?

Reply:
i think this was a change the middle of the refactor, changing it back